prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -161,6 +165,37 @@ def canonicalize_dataset_using_columns_for_split(
             {"input_col": input_col, "output_col": output_col}
         )
 
+    @staticmethod
+    def automatic_column_selection(
+        instruction: str, dataset_name: str, dataset_columns: str, example_rows: dict
+    ) -> tuple[list[str], str]:
+        """Find appropriate input and output columns for a given dataset and tasks."""
+        prompt = construct_prompt_for_column_selection(
+            instruction, dataset_name, dataset_columns, example_rows
+        )
+        required_keys = ["input", "output"]
+        optional_keys = ["ambiguous", "irrelevant"]
+
+        response = parse_prompt_to_fields(prompt, required_keys, optional_keys)
+        input_columns = response["input"]
+        output_column = response["output"]
+
+        if len(input_columns) < 1 or len(output_column) != 1:
+            raise RuntimeError(
+                "Input columns length was less than 1 or output column length was not 1"
+            )
+
+        incorrect_columns = [
+            col for col in input_columns + output_column if col not in dataset_columns
+        ]
+        if len(incorrect_columns) > 0:
+            raise RuntimeError(
+                f"One or more columns ({incorrect_columns}) were output that were "
+                f"not in the list of columns in the dataset ({dataset_columns})."
+            )  # TODO: potentially could also mean that the prompt needs to be updated
```

Comment:
I think we'd better not leave TODOs in the codes but add an issue instead.

Reply:
Yes you're right. I think my intention was more to point out that *if* this issue comes up, then it is more indicative of an error on our end, and not in general. Do you think it would make sense to just log the error as a warning then?

Reply:
Do agree so! Adds a warning logging definitely!

Reply:
I actually don't think that we need any action here. It's always the case that we might be able to find better prompts, so I don't think we need to create an issue until it actually becomes a problem. Maybe the TODO can just be removed.

```suggestion
            )
```